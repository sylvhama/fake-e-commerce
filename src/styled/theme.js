const theme = {
  color: {
    primary: '#ea352d',
    appBackground: '#f5f5f5',
    white: '#fff',
    lightGrey: '#ddd',
    grey: '#333',
    darkGrey: '#212121'
  },
  font: { h1: '2.5rem', h2: '2rem', h3: '1.5rem', p: '1rem' },
  size: {
    categories: '40px',
    footer: '60px',
    container: {
      reg: '940px',
      lg: '1020px'
    }
  },
  spacing: {
    sm: '8px',
    reg: '16px',
    lg: '24px'
  },
  radius: '4px'
};

export default theme;
