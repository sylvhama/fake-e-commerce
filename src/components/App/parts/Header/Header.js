import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import Container from '../../../shared/Container';
import Logo from '../../../shared/Logo';

const Header = styled.header`
  background-color: ${props => props.theme.color.white};
`;

export default props => (
  <Header {...props}>
    <Container maxWidth="lg">
      <Link to="/">
        <Logo />
      </Link>
    </Container>
  </Header>
);
