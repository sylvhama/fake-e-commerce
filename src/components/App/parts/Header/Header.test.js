import React from 'react';
import 'jest-styled-components';

import { renderWithTheme, wrapWithRouter } from '../../../../tests/helpers';
import theme from '../../../../styled/theme';

import Header from './Header';

test('Header renders correctly', () => {
  const rendered = renderWithTheme(
    wrapWithRouter(<Header>Header</Header>)
  ).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('background-color', theme.color.white);
});
