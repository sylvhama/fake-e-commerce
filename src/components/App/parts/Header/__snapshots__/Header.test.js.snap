// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c1 {
  margin: 0 auto;
  padding: 16px 8px 0;
  max-width: 1020px;
}

.c0 {
  background-color: #fff;
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        height="50"
        width="182"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#dedede"
          height="46"
          width="178"
          x="2"
          y="2"
        />
        <text
          alignmentBaseline="middle"
          fill="#555555"
          fontSize="18"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          Logo SVG
        </text>
      </svg>
    </a>
  </div>
</header>
`;
